app/containers/react-native-web/artist/grid.tsx

```diff
@@ -0,0 +1,373 @@
+          const aspectRatio = artwork.image.aspect_ratio || 1 // Ensure we never divide by null/0
+          // Invert the aspect ratio so that a lower value means a shorter section.
+          sectionRatioSums[sectionIndex] += (1 / aspectRatio)
+        }
+      }
+    }
+
+    return sectionedArtworks
+  }
+
+  renderSections() {
+    const spacerStyle = {
+      height: this.props.itemMargin,
+    }
+
+    const sectionedArtworks = this.sectionedArtworks()
+    const sections = []
+    for (let i = 0; i < this.props.sectionCount; i++) {
+      const artworkComponents = []
+      const artworks = sectionedArtworks[i]
+      for (let j = 0; j < artworks.length; j++) {
+        const artwork = artworks[j]
+        artworkComponents.push(<Artwork artwork={artwork} key={`artwork-${j}-${artwork.__id}`} />)
+        // Setting a marginBottom on the artwork component didn’t work, so using a spacer view instead.
+        if (j < artworks.length - 1) {
+          artworkComponents.push(
+            <View style={spacerStyle} key={'spacer-' + j + '-' + artwork.__id} accessibilityLabel="Spacer View" />
+          )
+        }
+      }
+
+      const sectionSpecificStyle = {
+        width: this.state.sectionDimension,
+        marginRight: (i === this.props.sectionCount - 1 ? 0 : this.props.sectionMargin),
+      }
+
+      sections.push(
+        <View style={[styles.section, sectionSpecificStyle]} key={i} accessibilityLabel={'Section ' + i} >
+          {artworkComponents}
+        </View>
+      )
+    }
+    return sections
+  }
+
+  // Lifted pretty much straight from RN’s ListView.js
+  onScroll = (event: React.NativeSyntheticEvent<React.NativeScrollEvent>) => {
+    const scrollProperties = event.nativeEvent
+    const contentLength = scrollProperties.contentSize.height
+    if (contentLength !== this._sentEndForContentLength) {
+      const offset = scrollProperties.contentOffset.y
+      const visibleLength = scrollProperties.layoutMeasurement.height
+      const distanceFromEnd = contentLength - visibleLength - offset
+      if (distanceFromEnd < PageEndThreshold) {
+        this._sentEndForContentLength = contentLength
+        this.fetchNextPage()
+      }
+    }
+  }
+
+  render() {
+    // const artworks = this.state.sectionDimension ? this.renderSections() : null
+    const artworks = this.renderSections()
+    return (
+      <ScrollView onScroll={this.onScroll}
+        scrollEventThrottle={50}
+        // onLayout={this.onLayout}
+        // scrollsToTop={false}
+        accessibilityLabel="Artworks ScrollView" >
+        <View style={styles.container} accessibilityLabel="Artworks Content View" >
+          {artworks}
+        </View>
+        {this.state.fetchingNextPage ? <Spinner style={styles.spinner} /> : null}
+      </ScrollView>
+    )
+  }
+}
+
+interface Styles {
+  container: ViewStyle,
+  section: ViewStyle,
+  spinner: ViewStyle,
+}
+
+const styles = StyleSheet.create<Styles>({
```

Comment:
this is a nice idea, I assume then it does correct autocompletion inside the children?